src/main/java/duke/Deadline.java

```diff
@@ -0,0 +1,50 @@
+package duke;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * This class is a deadline task. It contains
+ * the description of the task and the time
+ * the task is due.
+ */
+public class Deadline extends Task {
+
+    private LocalDate by;
```

Comment:
The variable name `by` might not be very clear to readers regarding what value it holds. A suggestion would be `deadline`.